### ISSUE TITLE
[TabLayout] TabLayoutMediator.detach() method causes NPE

### ISSUE THREAD

**Issue by joshuamking-driveclutch** [material-components/material-components-android#804]
**Description:** NPE is caused due to a call to `this.adapter.unregisterAdapterDataObserver` with a null `this.pagerAdapterObserver`. This likely means that I'm calling it twice or calling it before calling `attach`, but there's no indication that calling this method can result in a NPE.

**Expected behavior:** It should catch the NPE within the `detach` method.

**Material Library version:** 1.2.0-alpha01
No changes regarding `TabLayout` that should affect this from `1.2.0-alpha01...1.2.0-alpha02`

**Comment by wcshi** [comment]
consider adding null check here:
https://github.com/material-components/material-components-android/blob/6efedee7574d472fc86304647055bdbee5cadb8b/lib/java/com/google/android/material/tabs/TabLayoutMediator.java#L145

**Comment by joshuamking-driveclutch** [comment]
Yea I've accommodated for a potential NPE in my code, but I feel like this is something that the library should do a null check for.

**Comment by wcshi** [comment]
A change has been submitted internally and will be synced out soon.
